mapreduce/reducer.js

```diff
@@ -0,0 +1,39 @@
+const AWS = require("aws-sdk");
+const uuid = require("uuid");
+
+const dynamoDb = new AWS.DynamoDB.DocumentClient();
+
+module.exports.reducer = async (event, context, callback) => {
+  const data = JSON.parse(event.body);
+  const timestamp = new Date().getTime();
+
+  let reduceData = [];
+  let emit = emittedData => reduceData.push(emittedData);
+  eval(data.reduceFunction);
+  let doReduce = () =>
+    new Promise((resolve, reject) => {
```

Comment:
Why is this wrapped in a promise? It doesn't seem to be necessary

Reply:
I need to remove it. I had this everywhere because I’m bad at async programming lol